syntax = "proto3";
package proto;
option go_package = "./proto";
import "module.proto";

message GetRequest {
  string plugin_id = 1;
  string user_id = 2;
  string project_id = 3;
  string key = 4;
}

message GetResponse {
  bytes value = 1;
}

message SetRequest {
  string plugin_id = 1;
  string user_id = 2;
  string project_id = 3;
  string key = 4;
  bytes value = 5;
}

message DeleteRequest {
  string plugin_id = 1;
  string user_id = 2;
  string project_id = 3;
  string key = 4;
}

service Store {
  rpc Get(GetRequest) returns (GetResponse);
  rpc Set(SetRequest) returns (Empty);
  rpc Delete(DeleteRequest) returns (Empty);
}